torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -2273,47 +2306,55 @@ def create_wrapped_node(
     ):
         from .._trace_wrapped_higher_order_op import TransformGetItemToIndex
 
-        tx: InstructionTranslator = tx
-
-        def create_scalar():
-            return query.call_method(
-                tx,
-                "new_empty",
-                (VariableTracker.build(tx, []),),
-                {
-                    "dtype": VariableTracker.build(tx, torch.int32),
-                },
+        @_make_inlinable
+        def _create_scalar(query, dtype=torch.int32, requires_grad=False):
+            return query.new_empty([], dtype=dtype, requires_grad=requires_grad)
+
+        @_make_inlinable
+        def _create_scalars(query):
+            # list comprehension is not supported for inlinig yet
+            return (
+                _create_scalar(query),
+                _create_scalar(query),
+                _create_scalar(query),
+                _create_scalar(query),
             )
 
-        with discard_graph_changes(tx):
-            bhmn = [create_scalar() for _ in range(4)]
-            if fn_name == "score_mod":
-                scores_require_grad: bool = query.requires_grad
-                score = query.call_method(
-                    tx,
-                    "new_empty",
-                    (VariableTracker.build(tx, []),),
-                    {"requires_grad": VariableTracker.build(tx, scores_require_grad)},
+        if fn_name == "score_mod":
+
+            @_make_inlinable
+            def _fn(query, fn):
+                score, *_ = torch.ops.higher_order.speculate_subgraph(
+                    _create_scalar, (query, query.dtype, query.requires_grad)
```

Comment:
Could we have not used speculate_subgraph here?  
For example, I'm thinking this is just
```
UserDefinedFunction(_create_scalar).call_function(query, query.dtype, query.requires_grad)
```
What's the difference?

Reply:
Nvm I see what's going on